tests/microos/live_patching.pm

```diff
@@ -0,0 +1,112 @@
+#           3) Register Live Patching module if not done already
+#           4) Optionally add a kernel update repo
+#           5) Update system and reboot
+#           6) Get new kernel version and compare it
+#
+# Maintainer: qa-c team <qa-c@suse.de>
+
+use base "opensusebasetest";
+use strict;
+use warnings;
+use testapi;
+use transactional;
+use utils qw(zypper_call ensure_ca_certificates_suse_installed);
+
+sub register_lp_module {
+    my $arch        = get_required_var('ARCH');
+    my $regcode     = get_required_var('SCC_REGCODE');
+    my $sle_version = get_required_var('VERSION_SLE');
+    my $extensions  = script_output('SUSEConnect --list-extensions');
+    record_info('Extensions', $extensions);
+    unless ($extensions =~ m/Live Patching.*Activated/) {
+        record_info('Register', 'Registering module "sle-module-live-patching"');
+        trup_call("register -p sle-module-live-patching/$sle_version/$arch -r $regcode");
+        check_reboot_changes;
+        $extensions = script_output('SUSEConnect --list-extensions');
+        record_info('SUSEConnect', script_output('SUSEConnect --status-text'));
+        record_info('Extensions',  $extensions);
+    }
+    die('There was a problem activating the Live Patching module') unless ($extensions =~ m/Live Patching.*Activated/);
+    zypper_call '--gpg-auto-import-keys ref';
+}
+
+sub configure_lp {
+    # ensure sle-module-live-patching-release is installed
+    assert_script_run('rpm -q sle-module-live-patching-release');
+    if (script_output('zypper patterns --installed-only') !~ 'lp_sles') {
+        record_info('Pattern', 'Installing pattern lp_sles');
+        trup_call('pkg install -t pattern lp_sles');
+        check_reboot_changes;
+    }
+    record_info('Install', 'Installing kernel-default-livepatch');
+    trup_call('pkg install --oldpackage kernel-default-livepatch');
+    check_reboot_changes;
+    assert_script_run("sed -i 's/multiversion =.*/multiversion = provides:multiversion(kernel)/' /etc/zypp/zypp.conf");
+    assert_script_run("sed -i \'s/multiversion.kernels =.*/multiversion.kernels = latest/g\' /etc/zypp/zypp.conf");
+    assert_script_run("echo \"LIVEPATCH_KERNEL='always'\" > /etc/sysconfig/livepatching");
+    upload_logs('/etc/zypp/zypp.conf',         failok => 1);
+    upload_logs('/etc/sysconfig/livepatching', failok => 1);
+}
+
+sub setup_optional_repo {
+    # Option to use a test repository with an existing kernel update.
+    # This is useful if regular channels don't provide any kernel update
+    # and we want to force testing this feature
+    my $test_repo = get_var('SLE_MICRO_KERNEL_REPO');
+    if ($test_repo) {
+        ensure_ca_certificates_suse_installed;
+        record_info('Repo', "Adding repository $test_repo");
+        zypper_call "ar $test_repo";
+        zypper_call '--gpg-auto-import-keys ref';
+    }
+}
+
+sub run {
+    my ($self) = @_;
+
+    select_console 'root-console';
```

Comment:
`select_serial_terminal` fits better as LP and also `virtio console` are supported on `x86_64` and `s390x`